### PR TITLE
Fix comments in SOA optimization commit. Remove raw pointer.

### PR DESCRIPTION
**Pull request by dmitry741**
Fix some comments from the library architect. The main goal is removing raw pointer from SOANT class and using instead it safe wrapped up pointer.

### CODE REVIEW

**Comment by dmitry741** [comment]
make test

**Comment by dmitry741** [comment]
make test

**Comment by dmitry741** [comment]
make test

File: include/data_management/data/soa_numeric_table.h [hunk 1]
```diff
@@ -259,14 +250,64 @@ class DAAL_EXPORT SOANumericTable : public NumericTable
     }
 
 protected:
+    /**
+     *  <a name="DAAL-CLASS-DATA_MANAGEMENT__WRAPPEDRAWPOINTER"></a>
+     *  \brief   Class that provides functionality of deep copy.
+     */
+    class WrappedRawPointer
```
Comment (averbukh) [inline include/data_management/data/soa_numeric_table.h]: Coding style violated: declaration order

File: include/data_management/data/soa_numeric_table.h [hunk 2]
```diff
@@ -259,14 +250,64 @@ class DAAL_EXPORT SOANumericTable : public NumericTable
...
+                daal::services::daal_free(_arrOffsets);
+                _arrOffsets = NULL;
+                _count      = 0;
+            }
+        }
+
+        WrappedRawPointer() : _arrOffsets(NULL), _count(0) {};
+        ~WrappedRawPointer() { deallocate(); }
+
+        DAAL_INT64 * get() const { return _arrOffsets; }
```
Comment (averbukh) [inline include/data_management/data/soa_numeric_table.h]: const function returns pointer to non-const memory. Looks like a bug.

File: include/data_management/data/soa_numeric_table.h [hunk 3]
```diff
@@ -259,14 +250,64 @@ class DAAL_EXPORT SOANumericTable : public NumericTable
...
+
+        WrappedRawPointer & operator=(WrappedRawPointer const & wrapper)
```
Comment (averbukh) [inline include/data_management/data/soa_numeric_table.h]: Looks like it's simple to provide strong guarantee (do not modify state in th case of an error).

File: include/data_management/data/soa_numeric_table.h (hunk 1 above)
Comment (dmitry741) [reply include/data_management/data/soa_numeric_table.h]: done

File: include/data_management/data/soa_numeric_table.h (hunk 2 above)
Comment (dmitry741) [reply include/data_management/data/soa_numeric_table.h]: done

File: include/data_management/data/soa_numeric_table.h (hunk 3 above)
Comment (dmitry741) [reply include/data_management/data/soa_numeric_table.h]: done

File: include/data_management/data/soa_numeric_table.h [hunk 4]
```diff
@@ -450,30 +507,29 @@ class DAAL_EXPORT SOANumericTable : public NumericTable
         /* compute offsets */
         for (size_t i = 0; i < ncols; ++i)
         {
-            char * pv      = (char *)(_arrays[i].get());
-            _arrOffsets[i] = (DAAL_INT64)(pv - ptrMin);
-            DAAL_ASSERT(_arrOffsets[i] >= 0)
+            char * pv = (char *)(_arrays[i].get());
+            DAAL_ASSERT(static_cast<DAAL_UINT64>(pv - ptrMin) <= MaxVal<long long>::get())
```
Comment (ntfshard) [inline include/data_management/data/soa_numeric_table.h]:
```suggestion
            DAAL_ASSERT(static_cast<DAAL_UINT64>(pv - ptrMin) <= static_cast<DAAL_UINT64>(MaxVal<long long>::get()))
```

**Comment by dmitry741** [comment]
make test

**Comment by dmitry741** [comment]
make test

**Comment by dmitry741** [comment]
make test

**Pull request merged**